server/src/graphql/resolvers/doctor.resolver.ts

```diff
@@ -44,26 +46,48 @@ export default class DoctorResolver {
   }
 
   // 2. doctor login - check if doctor exists, check password, return doctor id or null
-  @Query(() => Int, {nullable: true})
+  @Query(() => String, {nullable: true})
   async loginDoctor (
     @Arg('email') email: string,
     @Arg('password') password: string,
-    @Ctx() { doctorRepo }: CustomContext
-  ): Promise<number|null> {
+    @Ctx() { doctorRepo , res}: CustomContext
+  ): Promise<string|null> {
     try {
       const doctor = await doctorRepo.findOne({email});
       if (!doctor) throw new Error (`${email}: No doctor found`);
 
       const passwordValid = await bcrypt.compare(password, doctor.password);
       if (!passwordValid) throw new Error (`${email}: Doctor invalid password`);
-      return doctor.id;
+
+      setRefreshToken(res, createRefreshToken(doctor.id)); // Set the refresh token inside a httpOnly cookie
+      return createAccessToken({id: doctor.id, isDoctor: true, language: doctor.language}); // Return access token
     } catch (e) {
       console.log(e);
       return null;
     }
   }
 
-  //3. Query - get one doctor by id, or docPublicCode
+  // 3. doctor login with JWT - if refresh token is found in cookies and it's valid, login automatically
+  @Query(() => String, {nullable: true})
+  async loginWithTokenDoctor (
+    @Ctx() { doctorRepo , req }: CustomContext
+  ): Promise<string|null> {
+    try {
+      const refreshToken = req.cookies['rtc'];
+      if (!refreshToken) return null;
+
+      const {id} = jwt.verify(refreshToken, process.env.REFRESH_TOKEN_SECRET as string) as { id:number };
+      const doctor = await doctorRepo.findOne({id});
+      if (!doctor) return null;
```

Comment:
On line 81, do we want to throw an error instead of just returning null? It might be more consistent with what you've done on line 56 above.